lib/azuremetadata/azuremetadata.py

```diff
@@ -15,25 +23,73 @@ def get_all(self):
         return result
 
     def get_instance_data(self):
-        # FIXME this is a stub while the CLI UX is being figured out
-
-        if self._api_version == '2019-08-15':
-            with open('fixtures/metadata-v2019-08-15-multiple-network-if.json', 'r') as json_file:
-                return json.load(json_file)
-        elif self._api_version == '2017-04-02':
-            with open('fixtures/metadata-v2017-04-02.json', 'r') as json_file:
-                return json.load(json_file)
-        else:
-            raise Exception("Unknown API version fixture exception")
+        return self._make_request(
+            "http://169.254.169.254/metadata/instance?api-version={}".format(quote(self._api_version))
+        )
 
     def get_attested_data(self):
-        # FIXME this is a stub while the CLI UX is being figured out
-
-        if self._api_version == '2019-08-15':
-            with open('fixtures/attested-data-v2019-08-15.json', 'r') as json_file:
-                return json.load(json_file)
-        elif self._api_version == '2018-10-01':
-            with open('fixtures/attested-data-v2018-10-01.json', 'r') as json_file:
-                return json.load(json_file)
-        else:
-            raise Exception("Unknown API version fixture exception (attested data)")
+        return self._make_request(
+            "http://169.254.169.254/metadata/attested/document?api-version={}".format(quote(self._api_version))
+        )
+
+    def get_disk_tag(self, device=None):
+        if not device:
+            device = self._find_root_device()
+
+        if not device:
+            return ''
+
+        try:
+            with open(device, 'rb') as fh:
+                fh.seek(65536)
+                return str(uuid.UUID(bytes_le=fh.read(16)))
+        except OSError as e:
+            print("An error occurred when reading disk tag:", file=sys.stderr)
+            print(str(e), file=sys.stderr)
+            return ''
+
+    @staticmethod
+    def _find_root_device():
+        hex_device_id = os.stat("/").st_dev
+        root_device_id = str(os.major(hex_device_id)) + ":" + str(os.minor(hex_device_id))
+
+        devices = glob.glob("/sys/block/*")
```

Comment:
glob pattern, single quotes please

Reply:
I think you are confused about how double quotes in Python. There's literally no difference.